### ISSUE TITLE
model.layers is empty after upgrading to 2.5.0

### ISSUE THREAD

**Issue by Earl7Fx3** [tensorflow/tensorflow#49976]
**System information**
- custom CNN model
- Linux Ubuntu 20.04
- TensorFlow installed from pip
- TensorFlow version: v2.5.0-rc3-213-ga4dfb8d1a71 2.5.0
- Python version: 3.8.5
- CUDA/cuDNN version: 11.1
- GPU model and memory: Quadro K5200

I upgraded Tensorflow  to 2.5.0 through pip so I could get support from Cuda 11.1.
After this upgrade, when I tried to access keras.optimizers.Adam, it wasn't there anymore, so I switched to tf.keras as it's recommended I see, but doing so I then realized that model.layers is now an empty list after building my model:

        self.model = keras.models.Model(inputs=[input_img], outputs=[outputs])
        self.model.compile(optimizer=keras.optimizers.Adam(),
                           loss="binary_crossentropy",
                           metrics=["accuracy"])
        print(self.model.layers) -> returns empty list

Tried updating tensorflow to a different computer, same issue with model.layers but I can still access keras.optimizers.Adam (this one has a slightly different setup as I'm using miniconda).

**Comment by jvishnuvardhan** [comment]
@Earl7Fx3 Did you train the model? Can you please share a simple standalone code to reproduce the issue? Thanks!

**Comment by Earl7Fx3** [comment]
Thanks for your reply!
I actually trained the model earlier, and had issues whiles trying to save it... Debugging it, I ended up figuring that model.layers was empty.
Here is a simple standalone code:

```python
import keras
import keras.layers as kl

def cv2d(input_tensor, n_filters, kernel_size=3, batchnorm=True):
    # first layer
    cv = kl.convolutional.Conv2D(filters=n_filters,
                                 kernel_size=(kernel_size, kernel_size),
                                 kernel_initializer="he_normal",
                                 padding="same")(input_tensor)
    if batchnorm:
        cv = kl.BatchNormalization()(cv)
    cv = kl.Activation("relu")(cv)
    # second layer
    cv = kl.convolutional.Conv2D(filters=n_filters,
                                 kernel_size=(kernel_size, kernel_size),
                                 kernel_initializer="he_normal",
                                 padding="same")(cv)
    if batchnorm:
        cv = kl.BatchNormalization()(cv)
    cv = kl.Activation("relu")(cv)
    return cv


def run():
    params = {'img_size': (256, 256, 1),
                  'n_filters': 16,
                  'batchnorm': True,
                  'dropout': 0.5}
    img_size = params['img_size']
    n_filters = params['n_filters']
    batchnorm = params['batchnorm']
    dropout = params['dropout']
    input_img = kl.Input(img_size, name='img')
    c1 = cv2d(input_img, n_filters=n_filters*1,
                                kernel_size=3, batchnorm=batchnorm)
    p1 = kl.pooling.MaxPooling2D((2, 2))(c1)
    p1 = kl.Dropout(dropout*0.5)(p1)
    
    c2 = cv2d(p1, n_filters=n_filters*2, kernel_size=3,
                                batchnorm=batchnorm)
    
    outputs = kl.convolutional.Conv2D(1, (1, 1), activation='sigmoid')(c2)
    model = keras.models.Model(inputs=[input_img], outputs=[outputs])
    model.compile(optimizer=keras.optimizers.Adam(),
                  loss="binary_crossentropy",
                  metrics=["accuracy"])
    print(model.layers)
```

which oututs:

```python
[<keras.engine.input_layer.InputLayer object at 0x7f2e96abb1c0>, <keras.layers.convolutional.Conv2D object at 0x7f2f2c176310>, <keras.layers.normalization_v2.BatchNormalization object at 0x7f2ec1d75e50>, <keras.layers.core.Activation object at 0x7f2e9c3009d0>, <keras.layers.convolutional.Conv2D object at 0x7f2e9c34ae20>, <keras.layers.normalization_v2.BatchNormalization object at 0x7f2e9c070370>, <keras.layers.core.Activation object at 0x7f2ec1349a60>, <keras.layers.pooling.MaxPooling2D object at 0x7f2ec1bf9dc0>, <keras.layers.core.Dropout object at 0x7f2ec1bf9820>, <keras.layers.convolutional.Conv2D object at 0x7f2e96fa45e0>, <keras.layers.normalization_v2.BatchNormalization object at 0x7f2e96fa4a00>, <keras.layers.core.Activation object at 0x7f2e9c304e50>, <keras.layers.convolutional.Conv2D object at 0x7f2e96aa55b0>, <keras.layers.normalization_v2.BatchNormalization object at 0x7f2e96aa5d30>, <keras.layers.core.Activation object at 0x7f2e9c4ff5b0>, <keras.layers.convolutional.Conv2D object at 0x7f2e9c4ff430>]
```

Replacing "import keras" by "import tensorflow.keras as keras" outputs an empty list.

**Comment by jvishnuvardhan** [comment]
@Earl7Fx3 I updated couple of lines and imported `keras` and `keras.layers`  from `tensorflow`. With those modiications, i don't see any difference in the output from `keras` or `tensorflow.keras`. As `tf.keras.` has no `convolutional` attribute,  I updated `Conv2D` layer in your code.  Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/28bcd8d72ff42b2177bff08354e3bbf4/untitled.ipynb).

```
from tensorflow import keras
from tensorflow.keras import layers as kl
```

Please verify once and close the issue if this was resolved for you. Thanks!

**Comment by Earl7Fx3** [comment]
I just checked, it now works with your modifications.
Thanks a lot for going through this!
